files/ansible/module_utils/test_result_parser/factory.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+    Base Factory class, with common logic for finding and loading models
+"""
+import os
+import re
+
+try:
+    import test_models.testmodel
+    from test_models.test_model_implementations import *
+except ImportError:
+    import ansible.module_utils.test_result_parser.test_models.testmodel
+    from ansible.module_utils.test_result_parser.test_models.test_model_implementations import *
+
+class TestFactory(object):
+    def __init__(self):
+        self.model_types = {
+                             'bibwtest_openmpi3_model': {'Bi-Directional Bandwidth Test': r'\s+OSU\sMPI\s(\w+[-]\w+\s\w+\s\w+)'},
+                             'bibwtest_model': {'Bidirectional BW Test': r'\s+RDMA_Write\s(\w+\s\w+\s\w+)'},
+                             'bwtest_model': {'BW Test': r'\s+RDMA_Write\s(\w+\s\w+)'},
+                             'bwtest_openmpi3_model': {'Bandwidth Test': r'\s+OSU\sMPI\s(\w+\s\w+)'},
+                             'latency_model': {'Latency Test': r'\s+RDMA_Write\s(\w+\s\w+)'},
+                             'latency_openmpi3_model': {'Latency Test': r'\s+OSU\sMPI\s(\w+\s\w+)'},
+                           }
+
+    def getTest(self, test_output):
```

Comment:
this method is useless. Either move the logic in find_model here, or remove this method and call find_model directly.